mmv1/products/securesourcemanager/BranchRule.yaml

```diff
@@ -0,0 +1,148 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'BranchRule'
+base_url: 'projects/{{project}}/locations/{{location}}/repositories/{{repository_id}}/branchRules?branch_rule_id={{branch_rule_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/repositories/{{repository_id}}/branchRules/{{branch_rule_id}}'
+immutable: true
+description: 'BranchRule is the protection rule to enforce pre-defined rules on designated branches within a repository.'
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/secure-source-manager/docs/overview'
+id_format: '{{repository}}/branchRules/{{branch_rule_id}}'
+import_format: ['projects/{{project}}/locations/{{location}}/repositories/{{repository_id}}/branchRules/{{branch_rule_id}}', '{{branch_rule_id}}']
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  actions: ['create', 'delete']
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+iam_policy: !ruby/object:Api::Resource::IamPolicy
+  parent_resource_attribute: 'repository_id'
+  method_name_separator: ':'
+  allowed_iam_role: 'roles/securesourcemanager.repoAdmin'
+  import_format: ['projects/{{project}}/locations/{{location}}/repositories/{{repository_id}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'secure_source_manager_branch_rule_basic'
+    primary_resource_id: 'default'
+    primary_resource_name: "fmt.Sprintf(\"tf-test-my-repository%s\",
+      context[\"random_suffix\"\
+      ])"
+    vars:
+      branch_rule_id: 'my-branchrule'
+      repository_id: 'my-repository'
+      prevent_desrtroy: 'true'
+    test_vars_overrides:
+      prevent_destory: 'false'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'secure_source_manager_branch_rule_with_fields'
+    primary_resource_id: 'default'
+    primary_resource_name: "fmt.Sprintf(\"tf-test-my-repository%s\",
+      context[\"random_suffix\"\
+      ])"
+    vars:
+      branch_rule_id: 'my-branchrule'
+      repository_id: 'my-repository'
+      prevent_desrtroy: 'true'
```

Comment:
Typo, also can you add a lifecycle prevent destroy to the instance and repository in the examples like in the other examples (I just added it)

Reply:
fixed.